### ISSUE TITLE
Parameter in path

### ISSUE THREAD

**Issue by fbernhard** [diosmosis/swagger-api-tester#1]
In the swagger file we define an API path like this:

```
/user/{user_id}/details
```

We then test with the following code:

```
test.get('/user/1/details').respondsWith(200)
```

This throws an exception:

```
Error: Path /user/1/details cannot be found in the swagger spec.
```

How can we test a path with a parameter in it? Thanks!

**Comment by diosmosis** [comment]
This should just work, I'll look into why it doesn't when I have some time. Would you be able to post the section in your swagger spec for that route?

**Comment by fbernhard** [comment]
Hi diosmosis,

Thanks for the quick reply and sorry for the delay. Please find below the details of a sample route, swagger file and test. Thanks for your help!

Express route:

```
  app.get('/api/test/user/:id/details', function(req, res) {
    res.send({
      id: req.params.id,
      name: "John Doe"
    })
  });
```

Swagger file:

```
swagger: "2.0"
basePath: /api/test
produces: 
  - application/json
schemes: 
  - http

host: localhost:8081
info: 
  description: "A test"
  title: "Test"
  version: "1.0.0"
definitions: 
  Error: 
    properties: 
      code: 
        format: int32
        type: integer
      message: 
        type: string
    type: object
  User: 
    type: object
    properties: 
      id:
        type: integer
        format: int32
        description: "The user id"
      name:
        type: string
        description: "The user name"
paths: 
  /user/{user_id}/details:
    get:
      description: |
        Loads user details
      tags: 
        - User
      parameters:
        - name: user_id
          description: "The user id"
          in: path
          required: true
          type: integer
          format: int32
      responses: 
        200: 
          description: "The user details"
          schema: 
            $ref: "#/definitions/User"
        default:
          description: "Unexpected error"
          schema: 
            $ref: "#/definitions/Error" 
```

Test:

```
const Tester = require('swagger-api-tester')

it('User test should run successfully', function(done) {
  const test = new Tester("./public/docs/test.yaml");

  describe('User details should return a JSON object', function() {
    test.get('/api/test/user/1/details').respondsWith(200, 'User');

    done();
  }); 
});
```

And finally the result:

```
$ mocha test/test.js 


  ✓ User test should run successfully 
  User details should return a JSON object
    1) should respond to GET /api/test/user/1/details correctly


  1 passing (28ms)
  1 failing

  1) User details should return a JSON object should respond to GET /api/test/user/1/details correctly:
     Error: Path /api/test/user/1/details cannot be found in the swagger spec.
      at getSwaggerResponse (/[...]/node_modules/swagger-api-tester/lib/request-test.js:90:11)
      at Context.<anonymous> (/[...]/node_modules/swagger-api-tester/lib/request-test.js:29:28)
      at next (native)
      at onFulfilled (/[...]/node_modules/co/index.js:65:19)
```

**Comment by diosmosis** [comment]
So there are three issues at play here, two are in the files you posted (due to the fact that I haven't written a README), and one is due to swagger-api-tester incorrectly setting the base URL for the API.

Your mocha test has two issues in it:

1) swagger-api-tester generates the `it(...)` calls, so you shouldn't need to use it.
2) the base path shouldn't be used in `test.xxx(...)` calls (so `/api/test` shouldn't be there)

An updated file would look like:

```
const Tester = require('swagger-api-tester')

describe('User test should run successfully', function() {
  const test = new Tester("./public/docs/test.yaml");
  // NOTE: you don't need to specify 'User' here either, since this is specified in the swagger docs.
  // Just .respondsWith(200) should work in this case too.
  test.get('/user/1/details').respondsWith(200, 'User');
});
```

I fixed the issue in swagger-api-tester and released version 1.0.4, can you try it out and see if it works for you?

**Comment by fbernhard** [comment]
Hi Diosmosis,

Thanks for clarification, it works well now. 

In particular, the following helped a lot, this is exactly what we need to extend the tests by plausibility checks of the data:

https://github.com/diosmosis/swagger-api-tester/blob/master/test/functional/example/petstore.test.js#L36
